ambry-clustermap/src/main/java/com.github.ambry.clustermap/Partition.java

```diff
@@ -193,7 +193,12 @@ public boolean equals(Object o) {
 
     Partition partition = (Partition) o;
 
-    return id == partition.id;
+    if (id == null ^ partition.id == null){
```

Comment:
Wasn't sure about JSONObject.getLong, but you're right: that throws an exception if the value is unavailable or not a long. Will remove the nullness check.